sceptre/connection_manager.py

```diff
@@ -105,6 +110,9 @@ def boto_session(self):
         with self._session_lock:
             self.logger.debug("Getting Boto3 session")
 
+            if self._is_session_expired():
```

Comment:
Consider this logic if `self.iam_role = None`. I think that with no iam_role, this will make `self._boto_session = None` everytime and recreate the session for every call. (i.e. the expiration will stay "in the past" (the initialised value) when logic flow uses the `else` block (lines 146-163)

Reply:
_is_session_expired() should only return true if iam_role is set, I have added a test that explicitly validates this :)

Reply:
with refactor suggested below - this if statement isn't needed.